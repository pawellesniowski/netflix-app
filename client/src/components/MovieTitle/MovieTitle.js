import React from 'react';

const MovieTitle = ({title}) => {
  return (
    <span className="movie-title">
      {title}
    </span>
  )
};

export default MovieTitle;
