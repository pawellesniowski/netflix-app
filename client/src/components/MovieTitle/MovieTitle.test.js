import React from 'react';
import { shallow } from 'enzyme';
import MovieTitle from './MovieTitle';

describe("MovieTitle Component with props", () => {
    let component;
    beforeEach(() => {
        component = shallow(<MovieTitle title="Rambo"/>);
    });

    it("Renders the container", () => {
        expect(component.find('[data-cy="movie-title"]')).toHaveLength(1);
    });

    it("Renders correct title", () => {
        expect(component.text()).toBe("Rambo");
    });

    test("Matches the snapshot", () => {
        expect(component).toMatchSnapshot();
    });
});

describe("MovieTitle  with NO props", () => {
    let component;
    beforeEach(() => {
        component = shallow(<MovieTitle />);
    });

    it("Renders the container", () => {
        expect(component.find('[data-cy="movie-title"]')).toHaveLength(1);
    });

    it("Renders correct title", () => {
        expect(component.text()).toBe("");
    });

    test("Matches the snapshot", () => {
        expect(component).toMatchSnapshot();
    });

});
