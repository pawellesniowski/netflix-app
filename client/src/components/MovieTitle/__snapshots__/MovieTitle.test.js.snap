// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitle  with NO props Matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`MovieTitle Component with props Matches the snapshot 1`] = `ShallowWrapper {}`;
